lib/ui/view/general_search/general_search_view.dart

```diff
@@ -0,0 +1,263 @@
+import 'dart:math';
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:zurichat/ui/shared/dumb_widgets/zuri_appbar.dart';
+import 'package:zurichat/ui/shared/shared.dart';
+import 'package:zurichat/utilities/constants/app_strings.dart';
+
+import 'package:zurichat/utilities/constants/text_styles.dart';
+import 'package:stacked/stacked_annotations.dart';
+import 'package:stacked/stacked.dart';
+import 'package:zurichat/utilities/internationalization/app_localization.dart';
+import 'general_search_view.form.dart';
+import 'general_search_viewmodel.dart';
+
+@FormView(fields: [FormTextField(name: 'search')])
+class GeneralSearchView extends StatelessWidget with $GeneralSearchView {
+  GeneralSearchView({Key? key}) : super(key: key);
+
+  String get needle => searchController.text.trim();
+
+  @override
+  Widget build(BuildContext context) {
+    final locale = AppLocalization.of(context)!;
+    return ViewModelBuilder<GeneralSearchViewModel>.reactive(
+      onModelReady: (model) => model.init(),
+      viewModelBuilder: () => GeneralSearchViewModel(),
+      initialiseSpecialViewModelsOnce: true,
+      builder: (context, model, child) => Scaffold(
+        appBar: ZuriAppBar(
+          isSearchBar: true,
+          hintText: locale.searchAll,
+          focusNode: searchFocusNode,
+          onTap: model.notifyListeners,
+          onChanged: (value) => model.notifyListeners(),
+          onEditingComplete: () {
+            FocusScope.of(context).unfocus();
+            model.search(needle);
+          },
+          searchController: searchController,
+          prefixIcon: needle.isEmpty
+              ? const Icon(Icons.search,
+                  color: AppColors.whiteColor, size: 20.0)
+              : IconButton(
+                  onPressed: () {
+                    FocusScope.of(context).unfocus();
+                    model.isShowingResults = false;
+                    searchController.clear();
+                    model.notifyListeners();
+                  },
+                  icon: const Icon(Icons.arrow_back_ios,
+                      color: AppColors.whiteColor, size: 16.0)),
+        ),
+        body: WillPopScope(
+          onWillPop: () async {
```

Comment:
You can carry the whole function to the view model and call notifyListeners there instead of here

Reply:
✌